controllers/client/round_tripper.go

```diff
@@ -2,20 +2,23 @@ package client
 
 import (
 	"crypto/tls"
+	"log/slog"
 	"net/http"
+	"strconv"
 
 	"github.com/grafana/grafana-operator/v5/embeds"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 type RoundTripperOnResponse func(requestMethod string, responseCode int)
 
 type instrumentedRoundTripper struct {
-	wrapped    http.RoundTripper
-	onResponse RoundTripperOnResponse
-	headers    map[string]string
+	wrapped http.RoundTripper
+	headers map[string]string
+	metrics []*prometheus.CounterVec
 }
 
-func NewInstrumentedRoundTripper(onResponse RoundTripperOnResponse, useProxy bool, tlsConfig *tls.Config) http.RoundTripper {
+func NewInstrumentedRoundTripper(useProxy bool, tlsConfig *tls.Config, metrics ...*prometheus.CounterVec) http.RoundTripper {
```

Comment:
yeah, passing them in as a list is another thought i had when doing this - seems like a good solution too!